### PR TITLE
Update WASM AOT guidance 8.0

### LINKED ISSUE #30151: WASM runtime in the .NET 8 enables EH and SIMD instructions by default.
WASM runtime in the .NET 8 enables EH and SIMD instructions by default.

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: fa3e99ba-cbe1-e8a1-6628-9b6eb39710a4
* Version Independent ID: dd777362-b92d-cac6-2343-a7aaf31d0bdf
* Content: [Tooling for ASP.NET Core Blazor](https://learn.microsoft.com/en-us/aspnet/core/blazor/tooling?view=aspnetcore-8.0)
* Content Source: [aspnetcore/blazor/tooling.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/blazor/tooling.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-blazor**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #30151

I'm glad you said something in spite of my working on it. This is looking much better.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/blazor/tooling.md](https://github.com/dotnet/AspNetCore.Docs/blob/a9e78de3573eeca7b9d1886bcab7efe7489a207e/aspnetcore/blazor/tooling.md) | [Tooling for ASP.NET Core Blazor](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/tooling?branch=pr-en-us-30152) |

### CODE REVIEW

File: aspnetcore/blazor/tooling.md [hunk 1]
```diff
@@ -690,37 +690,68 @@ The .NET WebAssembly build tools are based on [Emscripten](https://emscripten.or
...
+To enable AOT, set `<RunAOTCompilation>` to `true` in the app's project file (`.csproj`):
+
 ```xml
 <PropertyGroup>
   <RunAOTCompilation>true</RunAOTCompilation>
 </PropertyGroup>
 ```
 
-:::moniker-end
+To disable SIMD, for example when targeting old or mobile devices, add the `<RunAOTCompilation>` property set to `false`:
```
Comment (pavelsavara) [inline aspnetcore/blazor/tooling.md:701]: To disable SIMD, for example when targeting old browsers (on mobile devices), add the `<WasmEnableSIMD>` property set to `false`:

File: aspnetcore/blazor/tooling.md [hunk 2]
```diff
@@ -690,37 +690,68 @@ The .NET WebAssembly build tools are based on [Emscripten](https://emscripten.or
...
 
-:::moniker range=">= aspnetcore-7.0 < aspnetcore-8.0"
+```xml
+<PropertyGroup>
+  <RunAOTCompilation>true</RunAOTCompilation>
```
Comment (pavelsavara) [inline aspnetcore/blazor/tooling.md:705]: `RunAOTCompilation` should not be necessary anymore, right @radekdoulik ?

File: aspnetcore/blazor/tooling.md [hunk 3]
```diff
@@ -690,37 +690,68 @@ The .NET WebAssembly build tools are based on [Emscripten](https://emscripten.or
...
 </PropertyGroup>
 ```
 
 :::moniker-end
 
-:::moniker range=">= aspnetcore-7.0"
+:::moniker range=">= aspnetcore-7.0 < aspnetcore-8.0"
 
-To enable WebAssembly exception handling, use the `<WasmEnableExceptionHandling>` property in the app's project file (`.csproj`) with a value of `true`:
+When [ahead-of-time (AOT) compilation](xref:blazor/host-and-deploy/webassembly#ahead-of-time-aot-compilation) is used, [WebAssembly Single Instruction, Multiple Data (SIMD)](https://github.com/WebAssembly/simd/blob/master/proposals/simd/SIMD.md) is supported for all major browsers. SIMD can improve the throughput of vectorized computations by performing an operation on multiple pieces of data in parallel using a single instruction.
```
Comment (pavelsavara) [inline aspnetcore/blazor/tooling.md:730]: AOT is orthogonal concept to SIMD, even Mono interpreter (no-AOT build) could use SIMD & EH instructions.

Comment (guardrex) [reply aspnetcore/blazor/tooling.md:730]: I see ... sounds like I should split out SIMD into its own section.

What about EH? ... What is that related to (i.e., which section would that appear in)?

Comment (pavelsavara) [reply aspnetcore/blazor/tooling.md:730]: both EH and SIMD are instructions which make things faster.
both could be used with and without AOT.
They are both enabled in Net8.

So, I think that SIMD and EH sections are on the same level as AOT, not subsection.

File: aspnetcore/blazor/tooling.md (hunk 2 above)
Comment (pavelsavara) [reply aspnetcore/blazor/tooling.md:705]: I think RunAOTCompilation was necessary to use SIMD in Net7 as interpreter didn't use it at all. But that's not the case anymore.

Comment (guardrex) [reply aspnetcore/blazor/tooling.md:726]: Ok ... I re-organized everything on the latest commit.

The AOT section duplicates the content for 6.0/7.0 and 8.0+ until we hear back from @radekdoulik. 👂

File: aspnetcore/blazor/tooling.md [hunk 4]
```diff
@@ -675,20 +675,39 @@ For more information on template options, see the following resources:
...
 
 * For the **ASP.NET and web development** workload in the Visual Studio installer, select the **.NET WebAssembly build tools** option from the list of optional components.
 * Execute `dotnet workload install wasm-tools` in an administrative command shell.
 
+> [!NOTE]
+> .NET WebAssembly build tools for .NET 6 projects
+>
+> The `wasm-tools` workload installs the build tools for .NET 7 projects. However, the .NET 7 version of the build tools are incompatible with existing projects built with .NET 6. Projects using the build tools that must support both .NET 6 and .NET 7 must use multi-targeting.
+>
+> Use the `wasm-tools-net6` workload for .NET 6 projects when developing apps with the .NET 7 SDK. To install the `wasm-tools-net6` workload, execute the following command from an administrative command shell:
```
Comment (pavelsavara) [inline aspnetcore/blazor/tooling.md:688]: There is already `wasm-tools-net7` and `wasm-tools` is Net8 I think

Comment (guardrex) [reply aspnetcore/blazor/tooling.md:688]: I made an update for `wasm-tools` to say it applies to the latest release, but I don't know what this should say for .NET 6 projects. You may need to supply the exact line of text that makes sense.

You probably already figured out ..........

mccoy

### *Damn it Jim! I'm a doc author, not a compiler expert!*

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&mdash; Dr. Leonard H. "Bones" McCoy ([DeForest Kelley](https://www.imdb.com/name/nm0001420/)) [Star Trek](http://www.startrek.com/), ©1966-69 [Paramount Pictures](http://www.paramount.com)

# 🙈😆

Comment (guardrex) [reply aspnetcore/blazor/tooling.md:688]: Note in passing that the rest of the AOT content (and runtime relinking in a following section) is at ...

https://learn.microsoft.com/en-us/aspnet/core/blazor/host-and-deploy/webassembly?view=aspnetcore-7.0#ahead-of-time-aot-compilation

I think a long-range task is to combine the content from both docs into one new article in the *Fundamentals* node. I opened an issue to do the work at ...

**Combine and relocate WASM build tools content**
https://github.com/dotnet/AspNetCore.Docs/issues/30153

With the .NET 8 workload, which will go right into 24Q1, I won't reach that for a while tho 🏃.

**Review by pavelsavara — APPROVED** [review]
LGTM, I guess you could merge it and if Radek would have more comments about AOT it could be separate PR

**Comment by guardrex** [comment]
I have the duplicated AOT content, so I'll remove that for now.

@radekdoulik, I'm going to merge this after one more commit and go ahead and merge it to the live doc set.

Could you look it over for the two versions to see if any changes are needed for either set of remarks ...

* **.NET 7 version**: https://learn.microsoft.com/en-us/aspnet/core/blazor/tooling?view=aspnetcore-7.0&pivots=windows#net-webassembly-build-tools
* **.NET 8 version**: https://learn.microsoft.com/en-us/aspnet/core/blazor/tooling?view=aspnetcore-8.0&pivots=windows#net-webassembly-build-tools

**Pull request merged**